packages/chips/ChipSet.js

```diff
@@ -65,54 +73,39 @@ export default class ChipSet extends Component {
     return {
       hasClass: (className) => this.classes.split(' ').includes(className),
       setSelected: (chipId, selected) => {
-        const {selectedChipIds} = this.state;
-        if (selected) {
-          selectedChipIds.add(chipId);
-        } else {
-          selectedChipIds.delete(chipId);
-        }
-        this.setState({selectedChipIds});
+        const selectedChipIds = this.state.foundation.getSelectedChipIds();
+        this.setState({selectedChipIds}, () => {
+          this.props.handleSelect(selectedChipIds);
+        });
       },
+      removeChip: this.removeChip,
     };
   }
 
-  updateChipSelection(ids = this.state.selectedChipIds) {
+  initChipSelection() {
```

Comment:
```suggestion
  initChipSelection(selectedChipIds) {
    this.updateChipSelection(selectedChipIds);
    this.setState({hasInitialized: true});
  }

  updateChipSelection(selectedChipIds) {
    React.Children.forEach(this.props.children, (child) => {
      const {id} = child.props;
      const selected = selectedChipIds.indexOf(id) > -1;
      if (selected) {
        this.state.foundation.select(id);
      } else {
        this.state.foundation.deselect_(id);
      }
    });
  }
```

Reply:
Since we're now updating the foundation every time props updates, we have to make `deselect_` public in MDC Web